reframe/core/pipeline.py

```diff
@@ -761,12 +761,14 @@ def _setup_environ(self, environ):
         for k, v in self.variables.items():
             self._current_environ.set_variable(k, v)
 
```

Comment:
Please put a small comment here why we are doing this. Something like the following: "Temporarily load the test's environment to record the actual module load/unload sequence"